screen/loans/src/main/java/com/ivy/loans/loan/LoansScreen.kt

```diff
@@ -100,46 +103,88 @@ private fun BoxWithConstraintsScope.UI(
 
         Spacer(Modifier.height(8.dp))
 
-        for (item in state.loans) {
-            Spacer(Modifier.height(16.dp))
+        if (state.tabularLoanMode) {
+            val loans = if (state.selectedTab == LoanTab.PENDING) {
+                state.pendingLoans
+            } else {
+                state.completedLoans
+            }
+
+            for (item in loans) {
+                Spacer(Modifier.height(16.dp))
 
-            LoanItem(
-                displayLoan = item
-            ) {
-                nav.navigateTo(
-                    screen = LoanDetailsScreen(
-                        loanId = item.loan.id
+                LoanItem(
+                    displayLoan = item
+                ) {
+                    nav.navigateTo(
+                        screen = LoanDetailsScreen(
+                            loanId = item.loan.id
+                        )
                     )
+                }
+            }
+
+            if (loans.isEmpty()) {
+                Spacer(Modifier.weight(1f))
+
+                NoLoansEmptyState(
+                    emptyStateTitle = stringResource(R.string.no_loans),
+                    emptyStateText = stringResource(R.string.no_loans_description)
                 )
+
+                Spacer(Modifier.weight(1f))
+            }
+        } else {
+            for (item in state.loans) {
```

Comment:
Use LazyColumn instead of `for` because it's more efficient

Reply:
Done